services/venice-router/src/main/java/com/linkedin/venice/router/api/RouterExceptionAndTrackingUtils.java

```diff
@@ -30,12 +33,13 @@ public enum FailureType {
     SMART_RETRY_ABORTED_BY_MAX_RETRY_ROUTE_LIMIT, RESOURCE_NOT_FOUND, RETRY_ABORTED_BY_NO_AVAILABLE_REPLICA
   }
 
-  private static final StackTraceElement[] emptyStackTrace = new StackTraceElement[0];
-
   private static RouterStats<AggRouterHttpRequestStats> ROUTER_STATS;
 
   private static final Logger LOGGER = LogManager.getLogger(RouterExceptionAndTrackingUtils.class);
 
+  /** We do not fill in the stacktrace at all for "expected exceptions" (quota, etc) */
+  private static final boolean DO_NOT_FILL_IN_STACKTRACE = false;
```

Comment:
Kinda do not like the final `false` boolean.  How about create a new VeniceExceptionNoStackTrace() and call that instead of passing a final false boolean?

Reply:
Hmm... that could work in some cases but seems less flexible? Java does not have multiple inheritance, so if a subclass of `VeniceException` wants to achieve this behavior, they would still need the constructor param...?

Reply:
But I can get rid of the final variable and just pass false directly at the call site, if you prefer...?

Reply:
Yeah that would work too.

Reply:
Done.